lnrpc/routerrpc/router_backend.go

```diff
@@ -773,6 +774,21 @@ func (r *RouterBackend) UnmarshallHop(rpcHop *lnrpc.Hop,
 func (r *RouterBackend) UnmarshallRoute(rpcroute *lnrpc.Route) (
 	*route.Route, error) {
 
+	var err error
+
+	// Most routes we construct are from our node's perspective.
+	sourcePubKey := r.SelfNode
+
+	// Optionally override with supplied public key.
+	if rpcroute.SourcePubKey != "" {
+		sourcePubKey, err = route.NewVertexFromStr(
+			rpcroute.SourcePubKey,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("invalid source pubkey: %w", err)
+		}
+	}
+
 	prevNodePubKey := r.SelfNode
```

Comment:
`prevNodePubKey` is not necessarily selfnode anymore

Reply:
I think we should add tests for this change to make sure that things like this would be caught.

Reply:
gotta make sure that everywhere where `r.SelfNode` is used is still correct given this commit